Adds the Curse of the Clown, a new, funny spell for wizards!

trollbreeder opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
This adds a new spell to wizards, the Curse of the Clown!
It's a 2 cost touch spell that turns the target into a cluwne, dropping every piece of their clothing on the floor, giving them green, unremovable clown gear, and making them clumsy. 
Featuring Codersprites™️
Yes, you can self target with this spell. **No, this is not an april fools joke.**

## Demonstrations
Turning humans into miserable cluwnes (the big bold blue policy text has been removed, dont look at it)
https://streamable.com/96iu1
Messing with cyborgs
https://streamable.com/1jr27

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
There's a lot of murderbone tools the wizard gets, but sometimes you just wanna be silly without killing anyone. It's also a useful tool: it does not care about magic immunity, only if you're already clumsy or not, and it drops everything the target has on the floor, while also making them clumsy. Great for hit and run theft, and great for stripping the warden of his ability to fire weapons without firing at his foot. 
I hope this becomes a useful tool for wizards, especially those with steal objectives (Help! The wizard turned the CE into a clown and stole his magboots!)

## Changelog
:cl:
add: After a collaboration with ClownGov and the Wizard Federation, the Academy now teaches it's wizards a new silly spell.
add: Curse of the Clown has been added to spellbooks! It allows the invoker to turn people into clowns. Oh no.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

trollbreeder committed:
Implements the Curse of the Clown.

A dastardly spell that invokes the rage of a thousand clowns into your hands, turning whoever gets touched by it into a clumsy cluwne.

trollbreeder committed:
Merge remote-tracking branch 'upstream/master'

trollbreeder committed:
Merge branch 'master' into wizardclownspell

trollbreeder committed:
Resolves merge cornflakes on the action icon sprite

Note to self: update local before commiting changes

trollbreeder committed:
Changes the name of the clown hand

Before it was "the clown's revenge" now it's "the silly touch"

TiviPlus commented:
HONK

trollbreeder commented:
This is not an april fools joke, and i'll be sad if it gets closed because someone thought it was an april fools joke.

Shadowflame909 commented:
3 points for clowning?

You gotta go all the way if you're taking from goon.

Give them the green clown suits from the vendor and make it undroppable. Then you make em valid salid through mass announcement, similar to blood contract.

itseasytosee commented:
I think I've seem this one before

Archanial commented:
Why not make it pointed spell?

SkeletalElite commented:
Either reduce the cost or make it a pointed spell.

Spepbuthewranglesgithub commented:
make the clown gear unremovable and you have an A+ spell, humiliating the entirety of sec has never been better.

HomocidlePancakes commented:
Just port cluwns you nerd.

necromanceranne commented:
Honk!

trollbreeder commented:
I have taken your feedback and i will now be making the clown gear green and undroppable, which should buff the spell to a level comparable to Smite.

ATH1909 commented:
> I have taken your feedback and i will now be making the clown gear green and undroppable, which should buff the spell to a level comparable to Smite.

Make it match the cooldown and point cost of Smite/Flesh to Stone too, then. Compared to Flesh to Stone, it wouldn't instantly make your victims completely helpless for 5+ minutes (or instantly kill them like Smite does), but it would still have that nice ability to pierce magic immunity, which I think is a fair trade off.

Idea: Make using this spell on someone who's wearing a suit of human skin destroy the skin and spawn a clown simplemob (or a clown corpse), as an analog to the Smite+suit of human skin interaction. Maybe you could also add a Flesh to Stone+suit of human skin interaction that'd destroy the suit of human skin and create a statue (or maybe stone armor?)?

ATH1909 commented on code/modules/antagonists/wizard/equipment/spellbook.dm:
```diff
@@ -269,6 +269,13 @@
+	spell_type = /obj/effect/proc_holder/spell/targeted/touch/honk
+	category = "Offensive"
+	cost = 3
+	no_coexistance_typecache = /obj/effect/proc_holder/spell/targeted/touch/flesh_to_stone // there may only be one transformative touch spell
```
why tho

ATH1909 replied:
bro I wanna do pimp hand wizard (Smite+Curse of the Clown+Flesh to Stone+some other shit idk)

how could you do this to me

trollbreeder replied:
highlander rules

trollbreeder replied:
then again this could be pure misery (clown statue, be unable to do anything for 5 minutes then be a clown for the rest of the shift until someone sprays you with acid)

ATH1909 commented on code/modules/antagonists/wizard/equipment/spellbook.dm:
```diff
@@ -269,6 +269,13 @@
 	cost = 1
 
+/datum/spellbook_entry/honk
+	name = "Curse of the Clown"
```
*Curse of the Cluwne

trollbreeder replied:
Curse of the Clown is intentional.

trollbreeder committed:
Massively buffs curse of the clown.

Curse of the Clown now gives unremovable, horrifically green clown equipment, and is now a 2 points cost due to it's cooldown. The clown gear appears regardless if the species can have it or not. It's essentially a supercharged version of Curse of the Barnyard now.

trollbreeder commented:
Curse of the Clown has now been buffed. It's basically Curse of the Barnyard cranked up to maximum with a touch of Flesh to Stone.

trollbreeder committed:
Buffs cooldown

Maybe 2 minutes is too much

trollbreeder committed:
Clown suits can now "protect" from the Curse of the Clown

If you use Curse of the Clown on someone with a clown suit, the clown suit will turn into a simplemob clown. Suggestion by ATH1909.

trollbreeder committed:
Flesh to Stone is now compatible with Curse of the Clown

ATH1909 wanted to do a pimp hand gimmick and i said ok sure
also i was very clumsy in making this noncoexistance thing and forgot to add it to flesh to stone ahahahh

LordSaladin commented:
you should have went with a spell that turns people into pickles.

trollbreeder commented:
> you should have went with a spell that turns people into pickles.

Funniest shit i've ever seen

LordSaladin commented:
It will be the funniest pr oranges has ever seen

trollbreeder commented:
> It will be the funniest pr oranges has ever seen

But alas, this is no april fools PR as i've stated 2 times before

LordSaladin commented:
then do it as a different pr

Fikou commented on code/modules/spells/spell_types/touch_attacks.dm:
```diff
@@ -81,3 +81,21 @@
+	school = "transmutation"
+	charge_max = 800 // you have to wait 1 minute and 20 seconds because you're dropping all items and turning your target into a clown
+	clothes_req = TRUE
+	cooldown_min = 400 // 100 deciseconds reduction per rank
```
isnt 800 - 500 300

trollbreeder replied:
I am bad at math

trollbreeder replied:
Actually, can't you upgrade spells only 4 times?
800 - 100 - 100 - 100 - 100 = 400

ATH1909 replied:
what trollbreeder said

trollbreeder committed:
Get out of here policy

after a bit of banter maybe having cluwnes be valid at all times is a bit too far

trollbreeder commented:
Cluwne policy is gone. Crabrave.
They're now considered normal crewmembers in terms of escalation and aren't valid to kill.
This should prevent new players from going monkey see valid -> NEURON ACTIVATION -> monkey kill.

ATH1909 commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -106,6 +106,19 @@
 		to_chat(user, "<span class='notice'>Your Clown Mask has now morphed into [choice], all praise the Honkmother!</span>")
 		return 1
 
+/obj/item/clothing/mask/gas/cluwne_mask
```
why not make this a subtype of a normal clown mask?

trollbreeder replied:
Dont want to inherit the morphing of the clown mask. You will be jomker forever.

ATH1909 commented on code/modules/clothing/outfits/standard.dm:
```diff
@@ -174,6 +174,21 @@
+
+/datum/outfit/job/clown/cluwne/plasma
+	name = "Plasmacluwne"
+	head = /obj/item/clothing/head/helmet/space/plasmaman/clown
```
don't forget the mask

trollbreeder replied:
Inherited from the outfit/job/clown/cluwne type. Clown masks are internals masks.

ATH1909 commented on code/modules/clothing/outfits/standard.dm:
```diff
@@ -174,6 +174,21 @@
+	name = "Plasmacluwne"
+	head = /obj/item/clothing/head/helmet/space/plasmaman/clown
+	uniform = /obj/item/clothing/under/plasmaman/clown
+	r_hand = /obj/item/tank/internals/plasmaman/belt/full
```
try to put this on their belt or in their pocket, if possible

trollbreeder replied:
All the plasmaman outfits put the tank in their right hand.

ATH1909 commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -129,6 +129,15 @@
 
+/obj/item/clothing/shoes/clown_shoes/cluwne
+	name = "cluwne shoes"
+	desc = "The prankster's standard-issue cluwning shoes... wait, why can't i get out of them?"
```
*I

trollbreeder replied:
```suggestion
	desc = "The prankster's standard-issue cluwning shoes... wait, why can't I get out of them?"
```

ATH1909 commented on code/modules/spells/spell_types/godhand.dm:
```diff
@@ -110,3 +110,46 @@
+		H.visible_message("<span class='danger'>[H]'s [suit] animates and jumps off of [H.p_them()], turning into a clown!</span>")
+		H.dropItemToGround(suit)
+		qdel(suit)
+		new /mob/living/simple_animal/hostile/retaliate/clown(H.loc)
```
could we make this be a cluwne corpse instead of a living clown? the suit is made of DEAD flesh, after all.

alternatively, add a hidden feature where if you use this on a catatonic person or on a corpse that doesn't have a mind, they turn into a living clown simplemob.

or just ignore me, this isn't a critical feature

trollbreeder replied:
its a funny interaction like how ei nath reacts poorly to human skinsuits

ATH1909 commented on code/modules/spells/spell_types/godhand.dm:
```diff
@@ -110,3 +110,46 @@
+	item_state = "clown"
+
+/obj/item/melee/touch_attack/honk/afterattack(atom/target, mob/living/carbon/user, proximity)
+	if(!proximity || !ishuman(target) || !iscarbon(user))
```
proposal: make it so that using this on a slaughter demon turns them into a laughter demon

secondary proposal: make it so that using this on a bot (as in, a medibot, mulebot, secbot, etc.) turns them into a honkbot

tertriary proposal: make it so that using this on a cyborg (who isn't already a clownborg) switches them to the clown module.

trollbreeder replied:
YES YES YES THESE ARE 3 AWESOME IDEAS

trollbreeder committed:
Update code/modules/clothing/shoes/miscellaneous.dm

trollbreeder committed:
New interactions with borgs

Probably shitcode but this is funny as fuck

trollbreeder commented:
You can now turn borgs into clownborgs with this spell. God help us all

trollbreeder committed:
Gives curse of the clown'd borgs a law 0

I will regret commiting this

optimumtact commented:
The wizard does not need more gimmick spells.

wesoda25 commented:
Yes it does lol

fatted50 commented:
Yes it does

Fikou commented:
Yes it

TiviPlus commented:
yes it does

trollbreeder commented:
So that means wizard needs more killing tools?

cacogen commented:
seemed all right. as far as superfluous bullshit goes we've seen worse

trollbreeder commented:
T'wasn't just a gimmick, it was a good tool for stripping people when you're up close and personal, and is a good borg confuser.

TiviPlus commented:
It's combat viable and funny. I dont see why this was closed.
Wiz already has 500000 ways to kill you anyway so why is this spell a bad thing
